source/daplink/drag-n-drop/vfs_user.c

```diff
@@ -308,17 +311,56 @@ static uint32_t read_file_details_txt(uint32_t sector_offset, uint8_t *data, uin
     return pos;
 }
 
+// Text representation of each error type, starting from the rightmost bit
+static const char* error_type_names[] = {
+    "internal",
+    "transient",
+    "user",
+    "target",
+    "interface"
+};
+
 // File callback to be used with vfs_add_file to return file contents
 static uint32_t read_file_fail_txt(uint32_t sector_offset, uint8_t *data, uint32_t num_sectors)
 {
-    const char *contents = (const char *)error_get_string(vfs_mngr_get_transfer_status());
-    uint32_t size = strlen(contents);
+    uint32_t size = 0;
+    error_t status = vfs_mngr_get_transfer_status();
+    const char *contents = (const char *)error_get_string(status);
+    error_type_t type = error_get_type(status);
 
     if (sector_offset != 0) {
         return 0;
     }
 
-    memcpy(data, contents, size);
+    memcpy(data + size, error_prefix, strlen(error_prefix));
+    size += strlen(error_prefix);
+    memcpy(data + size, contents, strlen(contents));
+    size += strlen(contents);
+    data[size] = '\r';
```

Comment:
is there reason why not to use ´snprintf()` here ? it would simplify function a lot...

Reply:
The added size from using snprintf is too big for some configurations.

Reply:
Yeah thats true. I just saw [one place](https://github.com/ARMmbed/DAPLink/blob/eab079b8a1326f0a3647e3df33cf5dda13994a1f/source/daplink/daplink_debug.h#L80) where it was used but seems that it was behind some debug flag..